analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/SETestContext.cs

```diff
@@ -46,9 +51,46 @@ public SETestContext(string code, AnalyzerLanguage language, SymbolicCheck[] add
         public static SETestContext CreateCS(string methodBody, string additionalParameters, params SymbolicCheck[] additionalChecks) =>
             CreateCS(methodBody, additionalParameters, null, additionalChecks);
 
-        public static SETestContext CreateCS(string methodBody, string additionalParameters, string localFunctionName, params SymbolicCheck[] additionalChecks)
+        public static SETestContext CreateCS(string methodBody, string additionalParameters, string localFunctionName, params SymbolicCheck[] additionalChecks) =>
+            new(ClassCodeCS(methodBody, additionalParameters), AnalyzerLanguage.CSharp, additionalChecks, localFunctionName);
+
+        public static SETestContext CreateCSLambda(string methodBody, string lambdaFragment, params SymbolicCheck[] additionalChecks) =>
+            new(ClassCodeCS(methodBody, null), AnalyzerLanguage.CSharp, additionalChecks, null, lambdaFragment);
+
+        public static SETestContext CreateCSMethod(string method, params SymbolicCheck[] additionalChecks) =>
+            new($@"
+using System;
+
+public class Sample
+{{
+    {method}
+
+    private void Tag(string name, object arg) {{ }}
+}}", AnalyzerLanguage.CSharp, additionalChecks);
+
+        public static SETestContext CreateVB(string methodBody, params SymbolicCheck[] additionalChecks) =>
+            CreateVB(methodBody, null, additionalChecks);
+
+        public static SETestContext CreateVB(string methodBody, string additionalParameters, params SymbolicCheck[] additionalChecks)
         {
             var code = $@"
+Public Class Sample
```

Comment:
Shouldn't the VB and CSharp sample classes be kept synchronized? Otherwise, the tests are tough to read.

Reply:
The highlighting seems messed up, I did not touch the VB part. The VB is intentionally kept behind. We don't test everything for both languages, so we add only minimum stuff that is needed.